### PR TITLE
Add GitHub to Supervisor hack script

### PR DESCRIPTION
**Pull request by benjaminapetersen**
**Release note**:

```release-note

```

### CODE REVIEW

File: hack/prepare-supervisor-on-kind.sh [hunk 1]
```diff
@@ -290,6 +295,40 @@ EOF
...
+  # Make a Secret for the above GitHubIdentityProvider to describe the GitHub client configured.
+  # TODO: make clientID and clientSecret configurable, create & store values in our secret manager
+  cat <<EOF | kubectl apply --namespace "$PINNIPED_TEST_SUPERVISOR_NAMESPACE" -f -
+apiVersion: v1
+kind: Secret
+type: "secrets.pinniped.dev/github-client"
+metadata:
+  name: my-github-provider-client-secret
+stringData:
+  clientID: REDACTED
```
Comment (benjaminapetersen) [inline hack/prepare-supervisor-on-kind.sh:326]: Need to get an actual github idp to pipe into this.

File: hack/prepare-supervisor-on-kind.sh [hunk 2]
```diff
@@ -534,6 +592,12 @@ if [[ "$use_ad_upstream" == "yes" ]]; then
   echo
 fi
 
+if [[ "$use_github_upstream" == "yes" ]]; then
+  echo "    GitHub Username: your github username"
```
Comment (benjaminapetersen) [inline hack/prepare-supervisor-on-kind.sh:596]: May want to change what we say here.

File: hack/prepare-supervisor-on-kind.sh (hunk 1 above)
Comment (joshuatcasey) [reply hack/prepare-supervisor-on-kind.sh:326]: Let's make these env vars like `PINNIPED_TEST_SUPERVISOR_UPSTREAM_OIDC_CLIENT_ID` so we can pass them in from other scripts or dynamically

File: hack/prepare-supervisor-on-kind.sh (hunk 2 above)
Comment (joshuatcasey) [reply hack/prepare-supervisor-on-kind.sh:596]: I'm not sure we need this at all. There's no login flow that would ever require them to provide the CLI with their credentials.

**Review by joshuatcasey — APPROVED** [review]

**Pull request merged**